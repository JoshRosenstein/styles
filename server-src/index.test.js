/**
 * @jest-environment node
 */
// import styles from '../src'

describe('Configure', () => {
  /*
  it('removes vendor prefixing', () => {
    const myStyles = styles.create({prefix: false})
    const style = myStyles({
      flex: {display: 'flex'},
    })

    style('flex')
    expect(myStyles.extract()).toMatchSnapshot()
  })

  it('has customized vendor prefixing', () => {
    // adds prefixes to transform, but not flex
    const prefix = function(key, value, context) {
      if (typeof context !== 'number') throw 'fail'

      switch (key) {
        case 'transform':
          return true
        case 'disable':
          if (value !== 'flex') throw 'fail'
        // eslint-disable-next-line
        default:
          return false
      }
    }

    const myStyles = styles.create({prefix})
    const style = myStyles({
      flex: {display: 'flex', transform: 'translateX(30px)'},
    })

    style('flex')
    expect(myStyles.extract()).toMatchSnapshot()
  })

  it('adds nonce to style tags', () => {
    const myStyles = styles.create({nonce: 'EDNnf03nceIOfn39fn3e9h3sdfa'})
    const style = myStyles({
      flex: {display: 'flex'},
    })

    style('flex')
    expect(myStyles.extractTags()).toMatchSnapshot()
  })

  it('changes key to "css"', () => {
    const myStyles = styles.create({key: 'css'})
    const style = myStyles({
      flex: {display: 'flex'},
    })

    style('flex')
    expect(myStyles.extractTags()).toMatchSnapshot()
  })
})

describe('Usage', () => {
  it('extracts single style tag in prod', () => {
    const myStyles = styles.create({})
    const style = myStyles({
      flex: {display: 'flex'},
      btn: `
        border-radius: 1000px;
        background: blue;
        color: white;
      `,
    })

    style('flex')
    style('btn')
    expect(myStyles.extractTags()).toMatchSnapshot()
  })

  it('extracts global styles', () => {
    const myStyles = styles.create({})
    myStyles.global`
      :root {
        --hello: "world";
      } 
    `

    expect(myStyles.extractTags()).toMatchSnapshot()
  })

  it('extracts global variables', () => {
    const myStyles = styles.create({})
    myStyles.variables({
      colors: {
        blue: '#09a',
      },
    })

    expect(myStyles.extractTags()).toMatchSnapshot()
  })

  it('extracts theme variables', () => {
    const myStyles = styles.create({})
    myStyles.themes({
      dark: {
        colors: {
          primary: '#000',
        },
      },
      light: {
        colors: {
          primary: '#fff',
        },
      },
    })

    expect(myStyles.extractTags()).toMatchSnapshot()
    myStyles.theme('dark')
    expect(myStyles.extractTags()).toMatchSnapshot()
    myStyles.theme('light')
    expect(myStyles.extractTags()).toMatchSnapshot()
  })

  it('caches styles', () => {
    const myStyles = styles.create({})
    const style = myStyles({
      flex: {display: 'flex'},
      btn: `
        border-radius: 1000px;
        background: blue;
        color: white;
      `,
    })

    style('flex')
    style('btn')
    style('flex')
    style('btn')
    expect(myStyles.extractTags()).toMatchSnapshot()
  })
   */
  it('passes', () => {
    expect(true).toBe(true)
  })
})
