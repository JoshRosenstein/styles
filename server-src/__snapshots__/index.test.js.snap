// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Configure changes key to "css" 1`] = `"<style data-dash=\\"1ut9bc3\\" data-cache=\\"css\\">.css-1ut9bc3{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}</style>"`;

exports[`Configure has customized vendor prefixing 1`] = `"<style data-dash=\\"1vlpa1m\\" data-cache=\\"-ui\\">.-ui-1vlpa1m{display:flex;-webkit-transform:translateX(30px);-ms-transform:translateX(30px);transform:translateX(30px);}</style>"`;

exports[`Configure removes vendor prefixing 1`] = `"<style data-dash=\\"1ut9bc3\\" data-cache=\\"-ui\\">.-ui-1ut9bc3{display:flex;}</style>"`;

exports[`createStyleTagFromCache adds nonce to style tags 1`] = `"<style data-dash=\\"1ut9bc3\\" data-cache=\\"-ui\\" nonce=\\"EDNnf03nceIOfn39fn3e9h3sdfa\\">.-ui-1ut9bc3{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}</style>"`;

exports[`createStyleTagFromCache caches styles 1`] = `"<style data-dash=\\"1ut9bc3 1i986zm\\" data-cache=\\"-ui\\">.-ui-1ut9bc3{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}.-ui-1i986zm{border-radius:1000px;background:blue;color:white;}</style>"`;

exports[`createStyleTagFromCache extracts global styles 1`] = `"<style data-dash=\\"1r8lpmu\\" data-cache=\\"-ui\\">:root{--hello:\\"world\\";}</style>"`;

exports[`createStyleTagFromCache extracts global variables 1`] = `"<style data-dash=\\"1wmqlz7\\" data-cache=\\"-ui\\">:root{--ui-colors-blue:#09a;}</style>"`;

exports[`createStyleTagFromCache extracts theme variables 1`] = `"<style data-dash=\\"1e9tt6p 1gqix33\\" data-cache=\\"-ui\\">.-ui-dark-theme{--ui-colors-primary:#000;}.-ui-light-theme{--ui-colors-primary:#fff;}</style>"`;

exports[`createStyleTagFromCache extracts theme variables 2`] = `"<style data-dash=\\"1e9tt6p 1gqix33\\" data-cache=\\"-ui\\">.-ui-dark-theme{--ui-colors-primary:#000;}.-ui-light-theme{--ui-colors-primary:#fff;}</style>"`;

exports[`createStyleTagFromCache extracts theme variables 3`] = `"<style data-dash=\\"1e9tt6p 1gqix33\\" data-cache=\\"-ui\\">.-ui-dark-theme{--ui-colors-primary:#000;}.-ui-light-theme{--ui-colors-primary:#fff;}</style>"`;

exports[`createStyleTagFromString adds nonce to style tags 1`] = `"<style data-dash=\\"1ut9bc3\\" data-cache=\\"-ui\\" nonce=\\"EDNnf03nceIOfn39fn3e9h3sdfa\\">.-ui-1ut9bc3{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}</style>"`;

exports[`createStyleTagFromString caches styles 1`] = `"<style data-dash=\\"1ut9bc3 1i986zm\\" data-cache=\\"-ui\\">.-ui-1ut9bc3{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}.-ui-1i986zm{border-radius:1000px;background:blue;color:white;}</style>"`;

exports[`createStyleTagFromString extracts global styles 1`] = `"<style data-dash=\\"1r8lpmu\\" data-cache=\\"-ui\\">:root{--hello:\\"world\\";}</style>"`;

exports[`createStyleTagFromString extracts global variables 1`] = `"<style data-dash=\\"1wmqlz7\\" data-cache=\\"-ui\\">:root{--ui-colors-blue:#09a;}</style>"`;

exports[`writeStylesFromCache writes 1`] = `
Object {
  "filename": "-ui-8xbyu.css",
  "name": "-ui-8xbyu.css",
  "path": "./",
  "styles": ".-ui-1ut9bc3{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}.-ui-1i986zm{border-radius:1000px;background:blue;color:white;}",
}
`;

exports[`writeStylesFromCache writes custom hash 1`] = `
Object {
  "filename": "-ui-f8bCooDawg.css",
  "name": "-ui-f8bCooDawg.css",
  "path": "./",
  "styles": ".-ui-f8bCooDawg{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}",
}
`;

exports[`writeStylesFromCache writes custom key 1`] = `
Object {
  "filename": "css-mz4oaa.css",
  "name": "css-mz4oaa.css",
  "path": "./",
  "styles": ".css-1ut9bc3{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}.css-1i986zm{border-radius:1000px;background:blue;color:white;}",
}
`;

exports[`writeStylesFromCache writes custom name 1`] = `
Object {
  "filename": "foo.css",
  "name": "foo.css",
  "path": "./",
  "styles": ".-ui-1ut9bc3{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}.-ui-1i986zm{border-radius:1000px;background:blue;color:white;}",
}
`;

exports[`writeStylesFromString writes 1`] = `
Object {
  "filename": "-ui-8xbyu.css",
  "name": "-ui-8xbyu.css",
  "path": "./",
  "styles": ".-ui-1ut9bc3{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}.-ui-1i986zm{border-radius:1000px;background:blue;color:white;}",
}
`;

exports[`writeStylesFromString writes custom hash 1`] = `
Object {
  "filename": "-ui-f8bCooDawg.css",
  "name": "-ui-f8bCooDawg.css",
  "path": "./",
  "styles": ".-ui-f8bCooDawg{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}",
}
`;

exports[`writeStylesFromString writes custom key 1`] = `
Object {
  "filename": "css-mz4oaa.css",
  "name": "css-mz4oaa.css",
  "path": "./",
  "styles": ".css-1ut9bc3{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}.css-1i986zm{border-radius:1000px;background:blue;color:white;}",
}
`;

exports[`writeStylesFromString writes custom name 1`] = `
Object {
  "filename": "foo.css",
  "name": "foo.css",
  "path": "./",
  "styles": ".-ui-1ut9bc3{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;}.-ui-1i986zm{border-radius:1000px;background:blue;color:white;}",
}
`;
